modin/engines/ray/cudf_on_ray/frame/gpu_manager.py

```diff
@@ -0,0 +1,76 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import ray
+import cudf
+import pandas
+from .partition import cuDFOnRayFramePartition
+
+
+@ray.remote(num_gpus=1)
+class GPUManager(object):
+    def __init__(self, gpu_id):
+        self.key = 0
+        self.cudf_dataframe_dict = {}
+        self.gpu_id = gpu_id
+
+    def apply_non_persistent(self, first, other, func, **kwargs):
+        df1 = self.cudf_dataframe_dict[first]
+        df2 = self.cudf_dataframe_dict[other] if other else None
+        if not df2:
+            result = func(df1, **kwargs)
+        else:
+            result = func(df1, df2, **kwargs)
+        return self.store_new_df(result)
```

Comment:
apply_non_persistent is meant to just return the result without save the value in the cache right? If so, I think that this line should be `return result` instead of `return self.store_new_df(result)`

Reply:
Good point Luis. I'll work on that.